### PR TITLE
[fix] comma option enabled value parsing to use strings only

### PR DESCRIPTION
**Pull request by Om4ar**
in case of comma enabled .. parsing should check for strings only when using indexOf(',')

fixes: #328

### CODE REVIEW

File: dist/qs.js [hunk 1]
```diff
@@ -125,7 +125,7 @@ var parseValues = function parseQueryStringValues(str, options) {
             val = interpretNumericEntities(val);
         }
 
-        if (val && options.comma && val.indexOf(',') > -1) {
+        if (val && typeof val === 'string' && options.comma && val.indexOf(',') > -1) {
```
Comment (ljharb) [inline dist/qs.js]: please revert this; this file is only updated at release time.

File: lib/parse.js [hunk 2]
```diff
@@ -83,7 +83,7 @@ var parseValues = function parseQueryStringValues(str, options) {
             val = interpretNumericEntities(val);
         }
 
-        if (val && options.comma && val.indexOf(',') > -1) {
+        if (val && typeof val === 'string' && options.comma && val.indexOf(',') > -1) {
```
Comment (ljharb) [inline lib/parse.js]: can you please add a regression test?

File: main.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+const { parse } = require("./lib");
```
Comment (ljharb) [inline main.js]: by accident, i assume?

Comment (Om4ar) [reply main.js]: yes just removed it

**Comment by ljharb** [comment]
The test you added passes without your fix - can you tweak the test so it fails without the fix?

**Comment by Om4ar** [comment]
@ljharb  thank you 
i learned a lot from this

**Pull request merged**